libtransmission/torrent-magnet.cc

```diff
@@ -166,10 +141,9 @@ bool tr_torrentUseMetainfoFromFile(
     // tor should keep this metainfo
     tor->set_metainfo(*metainfo);
 
-    if (tor->incompleteMetadata != nullptr)
+    if (auto& m = tor->incomplete_metadata; m)
```

Comment:
This conditional isn't needed here; these four lines could just be `tor->incomplete_metadata.reset();`